tests/functional/object/mcg/test_noobaa_metrics.py

```
@@ -18,6 +19,39 @@
+            bucket_name=bucket_name,
+            key="Num Objects",
+        ):
+            last_seen = num_objects
+            if int(num_objects) == expected_count:
+                logger.info(f"Object count reached {num_objects}")
+                break
+    except TimeoutExpiredError:
+        assert False, (
+            f"Object count did not reach {expected_count} within {timeout}s, "
+            f"last observed: {last_seen}"
+        )
```

Ruff (0.15.21)

[warning] Assertion always fails, replace with `pytest.fail()`

(PT015)